infra/Pos/Network/Types.hs

```diff
@@ -198,6 +203,21 @@ topologyNodeType TopologyP2P{}         = NodeEdge
 topologyNodeType TopologyTraditional{} = NodeCore
 topologyNodeType TopologyAuxx{}        = NodeEdge
 
+-- | Type class which encapsulates something that has 'NodeType'.
+class HasNodeType ctx where
```

Comment:
I'd use reflection i guess.

Reply:
Why reflection? We already have `NetworkConfig` in reader context and we can deduce `NodeType` from it.